cranelift/codegen/src/opts/cprop.isle

```diff
@@ -24,12 +35,24 @@
       (if-let d (imm64_sdiv ty k1 k2))
       (iconst ty d))
 
+(rule (simplify_skeleton
+       (srem (iconst ty k1)
+             (iconst _ k2)))
```

Comment:
I think we need to match `ty` here too, otherwise the sign-extends in the helpers are wrong, right?